graphql/resolvers/users.js

```diff
@@ -170,81 +170,101 @@ module.exports = {
       return {
         ...res._doc,
         id: res._id,
-        token,
+        token
       };
     },
-    async deleteUser(_, { userId }) {
+    async deleteUser(_, { email, currentEmail }) {
+      let { errors, valid } = validateEmailInput(email);
+
+      if (!valid) {
+        throw new UserInputError("Errors.", {
+          errors
+        });
+      }
+
+      const loggedInUser = await User.findOne({
+        email: currentEmail
+      });
+
+      if (!loggedInUser) {
+        errors.general = "User not found";
+        throw new UserInputError("User not found", {
+          errors
+        });
+      }
+
+      if (!loggedInUser.permission.includes("admin")) {
+        valid = false;
+        errors.general = "Must be an admin to delete a user.";
+        throw new UserInputError("Must be an admin to delete a user.", {
+          errors
+        });
+      }
+
       try {
-        const user = await User.findById(userId);
+        const user = await User.findOne({ email });
         await user.delete();
         return "User deleted successfully";
```

Comment:
Can you return something other than a string? For example, after the try-catch, you can fetch all users and return them so we can see that the user got deleted. Everything else is fine :)

Reply:
done!